icechunk/src/repository.rs

```diff
@@ -269,6 +271,16 @@ impl Repository {
         &self.config
     }
 
+    pub(crate) async fn raise_if_invalid_snapshot_id(
```

Comment:
I really like this ! So much that I would have it as a free function that takes a `&dyn Storage`

Reply:
I'd put the function in the same file.